### DISCUSSION TITLE
LDM Notes for March 2nd, 2022

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#5860] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2022/LDM-2022-03-02.md
## Agenda

1. Open questions in `field`
    1. Initializers
    2. Property assignment in structs

**Comment by PathogenDavid** [comment]
A thought I had while reading the related discussion on the C# Discord, was it considered to have the field initialization be explicit to avoid the ambiguity? IE:

```csharp
record struct Rec(int Item)
{
    public int Item { get; set => value < 0 ? throw new Exception() : field = value; field = Item; }
}
```

It's not quite as terse, but I think it makes it more obvious you're assigning the field rather than invoking the setter.

Should the team ever chose to pursue them, I think this would also evolve more elegantly towards property-scoped fields and/or having a different type for the backing field. (Assuming it doesn't create a parsing nightmare.)
